server/sonar-server/src/main/java/org/sonar/server/plugins/ws/AvailablePluginsWsAction.java

```diff
@@ -0,0 +1,194 @@
+  }
+
+  private void writePluginUpdate(JsonWriter jsonWriter, PluginUpdate pluginUpdate) {
+    jsonWriter.beginObject();
+    Plugin plugin = pluginUpdate.getPlugin();
+
+    jsonWriter.prop(PROPERTY_KEY, plugin.getKey());
+    jsonWriter.prop(PROPERTY_NAME, plugin.getName());
+    jsonWriter.prop(PROPERTY_CATEGORY, plugin.getCategory());
+    jsonWriter.prop(PROPERTY_DESCRIPTION, plugin.getDescription());
+    jsonWriter.prop(PROPERTY_LICENSE, plugin.getLicense());
+
+    writeOrganisation(jsonWriter, plugin);
+
+    writeTermsAndConditions(jsonWriter, plugin);
+
+    writeRelease(jsonWriter, pluginUpdate.getRelease());
+
+    writeUpdate(jsonWriter, pluginUpdate);
+
+    jsonWriter.endObject();
+  }
+
+  private List<PluginUpdate> retrieveAvailablePlugins() {
+    return updateCenterFactory.getUpdateCenter(DO_NOT_FORCE_REFRESH).findAvailablePlugins();
+  }
+
+  private void writeRelease(JsonWriter jsonWriter, Release release) {
+    jsonWriter.name(OBJECT_RELEASE);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_VERSION, release.getVersion().toString());
+    jsonWriter.propDate(PROPERTY_DATE, release.getDate());
+    writeArchive(jsonWriter, release);
+    jsonWriter.endObject();
+  }
+
+  private void writeArchive(JsonWriter jsonWriter, Release release) {
+    jsonWriter.name(OBJECT_ARCHIVE);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_NAME, release.getFilename());
+    jsonWriter.prop(PROPERTY_URL, release.getDownloadUrl());
+    jsonWriter.endObject();
+  }
+
+  private void writeOrganisation(JsonWriter jsonWriter, Plugin plugin) {
+    jsonWriter.name(OBJECT_ORGANIZATION);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_NAME, plugin.getOrganization());
+    jsonWriter.prop(PROPERTY_URL, plugin.getOrganizationUrl());
+    jsonWriter.endObject();
+  }
+
+  private void writeTermsAndConditions(JsonWriter jsonWriter, Plugin plugin) {
+    jsonWriter.name(OBJECT_TERMS_AND_CONDITIONS);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_URL, plugin.getTermsConditionsUrl());
+    jsonWriter.endObject();
+  }
+
+  private void writeUpdate(JsonWriter jsonWriter, PluginUpdate pluginUpdate) {
+    jsonWriter.name(OBJECT_UPDATE);
+    jsonWriter.beginObject();
+    jsonWriter.prop(PROPERTY_STATUS, toJSon(pluginUpdate.getStatus()));
+
+    jsonWriter.name(ARRAY_REQUIRES);
+    jsonWriter.beginArray();
+    Release release = pluginUpdate.getRelease();
+    for (Plugin child : filter(transform(release.getOutgoingDependencies(), ReleaseToArtifact.INSTANCE), Plugin.class)) {
+      jsonWriter.beginObject();
+      jsonWriter.prop(PROPERTY_KEY, child.getKey());
+      jsonWriter.prop(AvailablePluginsWsAction.PROPERTY_NAME, child.getName());
+      jsonWriter.prop(PROPERTY_DESCRIPTION, child.getDescription());
+      jsonWriter.endObject();
+    }
+    jsonWriter.endArray();
+
+    jsonWriter.endObject();
+  }
+
+  private static String toJSon(PluginUpdate.Status status) {
+    switch (status) {
```

Comment:
to avoid exposing internal data in the WS (and having a regression the day some decides the values of the enum are too long)
also, it is less verbose => JSON message is shorter => better performance on the network and the client side